### ISSUE TITLE
[Bug] "Failed to set up"

### ISSUE THREAD

**Issue by CitizenRacer** [greghesp/ha-bambulab#39]
On the integrations page there is a black exclamation point in a red circle that says "Failed to set up" and a link to check the logs. Trying to reload the integration doesn't help. My printer no longer updates home-assistant

**To Reproduce**

**Expected behavior**
I expect the integration to connect to my printer.

**Screenshots**

**Device Type(please complete the following information):**
 - Printer: x1c
 - Firmware Version: X1C: 01.04.01.00 (Latest version) AMS: 00.00.05.96
 

**Home Assistant(please complete the following information):**
Home Assistant 2023.3.3 
Supervisor 2023.03.1 
Operating System 9.5 
Frontend 20230309.0 - latest 
Bambu integration c00119f

**Additional context**
logs:
2023-03-13 13:10:14.195 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration bambu_lab which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
File "/config/custom_components/bambu_lab/pybambu/bambu_client.py", line 32, in listen_thread
2023-03-13 13:11:09.979 ERROR (MainThread) [homeassistant.config_entries] Error setting 
![Screenshot 2023-03-13 at 1 13 16 PM](https://user-images.githubusercontent.com/14790049/224821684-a7f45add-a27e-4ead-9427-a7437751aadc.jpg)
up entry
[home-assistant_2023-03-13T20-11-41.147Z.log](https://github.com/greghesp/ha-bambulab/files/10961495/home-assistant_2023-03-13T20-11-41.147Z.log)
 00M00A2A1402560 for bambu_lab
File "/config/custom_components/bambu_lab/__init__.py", line 26, in async_setup_entry
File "/config/custom_components/bambu_lab/coordinator.py", line 67, in wait_for_data_ready
![Screenshot 2023-03-13 at 1 12 56 PM](https://user-images.githubusercontent.com/14790049/224821694-a662027b-d226-4451-9d38-ee9eb4d10d75.jpg)

**Comment by CitizenRacer** [comment]
deleting the printer and re-adding it worked to bring it back.

**Comment by CitizenRacer** [comment]
It did this before updating the integration. It continued doing it after I updated.

**Comment by CitizenRacer** [comment]
Not since I deleted and re-added the printer. I wish I could be more help!

**Comment by CitizenRacer** [comment]
Wait! It just did it again. The printer is off, I went to look at the integration and it says "failed to set up"

**Comment by CitizenRacer** [comment]
Logger: root
Source: custom_components/bambu_lab/pybambu/bambu_client.py:32
Integration: Bambu Lab ([documentation](https://github.com/greghesp/ha-bambulab), [issues](https://github.com/greghesp/ha-bambulab/issues))
First occurred: 7:33:35 AM (1 occurrences)
Last logged: 7:33:35 AM
Uncaught thread exception

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/usr/local/lib/python3.10/threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "/config/custom_components/bambu_lab/pybambu/bambu_client.py", line 32, in listen_thread
    self.client.connect(self.host, self._port, keepalive=5)
  File "/usr/local/lib/python3.10/site-packages/paho/mqtt/client.py", line 914, in connect
    return self.reconnect()
  File "/usr/local/lib/python3.10/site-packages/paho/mqtt/client.py", line 1044, in reconnect
    sock = self._create_socket_connection()
  File "/usr/local/lib/python3.10/site-packages/paho/mqtt/client.py", line 3685, in _create_socket_connection
    return socket.create_connection(addr, timeout=self._connect_timeout, source_address=source)
  File "/usr/local/lib/python3.10/socket.py", line 845, in create_connection
    raise err
  File "/usr/local/lib/python3.10/socket.py", line 833, in create_connection
    sock.connect(sa)
OSError: [Errno 113] Host is unreachable

**Comment by CitizenRacer** [comment]
Logger: homeassistant.bootstrap
Source: bootstrap.py:496
First occurred: 7:33:40 AM (1 occurrences)
Last logged: 7:33:40 AM
Waiting on integrations to complete setup: bambu_lab\

**Comment by CitizenRacer** [comment]
Logger: homeassistant.config_entries
Source: custom_components/bambu_lab/coordinator.py:67
Integration: Bambu Lab ([documentation](https://github.com/greghesp/ha-bambulab), [issues](https://github.com/greghesp/ha-bambulab/issues))
First occurred: 7:34:01 AM (1 occurrences)
Last logged: 7:34:01 AM
Error setting up entry 00M00A2A1402560 for bambu_lab

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 383, in async_setup
    result = await component.async_setup_entry(hass, self)
  File "/config/custom_components/bambu_lab/__init__.py", line 26, in async_setup_entry
    await coordinator.wait_for_data_ready()
  File "/config/custom_components/bambu_lab/coordinator.py", line 67, in wait_for_data_ready
    raise Exception('Failed to receive version response from printer in 30 seconds')
Exception: Failed to receive version response from printer in 30 seconds

**Comment by CitizenRacer** [comment]
Naive question: I don't leave my printer on all the time. Is it failing when the printer is off and it tries to connect? My system did restart when it updated HA Core.

**Comment by CitizenRacer** [comment]
Ahh, A suggestion then: Include this in the documentation and perhaps a more detailed error message? Is there a way I can kick off the integration to turn on when power is turned on? I'd love to add an automation "when the bambu plug is energized, kick the integration"

**Comment by CitizenRacer** [comment]
I turned on my printer and the integration hasn't connected, still in the failed state. Trying to force a reload didn't help either.

**Comment by CitizenRacer** [comment]
![Screenshot 2023-03-14 at 2 56 51 PM](https://user-images.githubusercontent.com/14790049/225149550-5ba0ff31-5187-4176-b303-be9773bf0f5a.jpg)


Just so we're clear, when you say reinitialize you mean clicking on "reload" on this menu right? 

My printer has been on for an hour and clicking "reload" still errors out with "failed to set up logs"

**Comment by CitizenRacer** [comment]
YEssir. It's static assigned by my Unifi router.

**Comment by AdrianGarside** [comment]
The printer off case was discussed a bunch in:
https://github.com/greghesp/ha-bambulab/issues/10
Restarting home assistant while the printer is off is currently not handled. You would have to restart home assistant after turning the printer back on to get the integation working.

To fix that I have reworked all this to allow the integration to initialize without the printer being on:
https://github.com/greghesp/ha-bambulab/pull/40

The wait_for_data_ready call that is failing in your repro logs is gone and I'm able to successfully restart home assistant (get all default values for the sensors currently - improving that will be done under a different issue), and then when you later turn on the printer everything populates. Although 'current stage' defaults to 'Printing' on my P1P - that's possibly a bug in the new P1P firmware since they just added that but if it's wrong on the X1 too then it's a bug in the integration.

**Comment by AdrianGarside** [comment]
There is something going on with reload - the integration doesn't support it currently but in my limited digging it seemed we weren't being told to do anything when the reload button is pressed. I was expecting to get called and then we'd need to close down the mqtt listener thread cleanly in order to allow reinitialization. But that's a separate issue from just not handling the printer being off.
